protocols/gossipsub/src/behaviour.rs

```diff
@@ -638,10 +636,6 @@ where
 
         let topic_hash = raw_message.topic.clone();
 
-        // If we are not flood publishing forward the message to mesh peers.
-        let mesh_peers_sent = !self.config.flood_publish()
-            && self.forward_msg(&msg_id, raw_message.clone(), None, HashSet::new())?;
```

Comment:
Why is this safe to remove? Does `forward_msg` not have a side-effect that we are now no longer performing?

Reply:
thanks for raising this Thomas, I was going to annotate it but you went ahead first :grin: 
basically `forward_msg` repeats all the peer counting that is done below besides the mesh peers, so I took that part of code and put it into the logic bellow when flood subbing is not enabled, this way we save repeated messages and are able to use `forward_msg` for `RpcOut::Forward`